src/Components/Explore/Explore.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react';
+import Post from '../Post/Post';
+import { FiFilter } from 'react-icons/fi';
+import { FaHeart } from 'react-icons/fa';
+import { IoIosArrowForward } from 'react-icons/io';
+import { GiSandsOfTime } from 'react-icons/gi';
+import { BiComment } from 'react-icons/bi';
+import { useSelector } from 'react-redux';
+import Spinner from '../Spinner/Spinner';
+import './Explore.css';
+
+const Explore = () => {
+  const { posts, loading } = useSelector((state) => state.post);
+  const [sortBy, setSortBy] = useState('latest');
+  const [sortedPosts, setSortedPosts] = useState([...posts]);
+
+  useEffect(() => {
+    setSortedPosts([...posts]);
+  }, [posts]);
+
+  if (sortBy === 'liked') {
+    sortedPosts.sort((a, b) => +b.likes.likeCount - +a.likes.likeCount);
+  } else {
```

Comment:
suggestion - I think any changes made to state variable should be done through the setState, sort is an impure function. but for now it does not require re-render so its working.

Reply:
@Vishnu-Aithal  Thank you for the suggestion, but any idea how else would I be able to attain the same results, since I am getting infinite rerenders if I do it through setState.

Reply:
try using useEffect with "sortBy" as dependency

Reply:
That was the first thing I did but it results into the bug @ApurvaSawant11 mentioned above

Reply:
now let it be, later you can try 
useEffect(() => {
    let newSortedPosts = [...posts];
    if (sortBy === "liked") {
        newSortedPosts.sort((a, b) => +b.likes.likeCount - +a.likes.likeCount);
    } else {
        newSortedPosts.sort((a, b) => +b.comments.length - +a.comments.length);
    }
    setSortedPosts(newSortedPosts);
}, [sortBy, setSortedPosts, posts]);